### DISCUSSION TITLE
Html-fund-me-cu- Couldn't Withraw

### DISCUSSION THREAD

**Discussion by 0xhung295** [Cyfrin/foundry-full-course-cu#3028] [Q&A] [answered]
I just recieved a issue involved withdraw() function when trying to withdraw money after fund:
This is the error:
'''
Error: execution reverted (unknown custom error) (action="estimateGas", data="0x579610db", reason=null, transaction={ "data": "0x3ccfd60b", "from": "0x452D94cA60902556Fd99a27839dFB468f36A7Bd1", "to": "0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512" }, invocation=null, revert=null, code=CALL_EXCEPTION, version=6.7.0)
    at makeError (ethers-6.7.esm.min.js:1:2824)
    at getBuiltinCallException (ethers-6.7.esm.min.js:1:246579)
    at AbiCoder.getBuiltinCallException (ethers-6.7.esm.min.js:1:248643)
    at BrowserProvider.getRpcError (ethers-6.7.esm.min.js:1:382690)
    at BrowserProvider.getRpcError (ethers-6.7.esm.min.js:1:430010)
    at ethers-6.7.esm.min.js:1:376477
'''
Even I tried with owner account, there're the same issue
Anyone have the same problem? thanks

**Comment by EngrPips** [comment]
I didn't encounter this problem when I withdrew with the owner account. Maybe check and confirm the `owner` address to be sure you are using the `owner` address to call the withdraw function.

**Reply by 0xhung295** [reply]
This is code of withdraw function:
'''solidity
function withdraw() public onlyOwner {
        for (
            uint256 funderIndex = 0;
            funderIndex < s_funders.length;
            funderIndex++
        ) {
            address funder = s_funders[funderIndex];
            s_addressToAmountFunded[funder] = 0;
        }
        s_funders = new address[](0);
        // Transfer vs call vs Send
        // payable(msg.sender).transfer(address(this).balance);
        (bool success, ) = i_owner.call{value: address(this).balance}("");
        require(success);
    }
'''

**Comment by KMean** [comment]
hey @0xhung295 can you share your withdraw function with us?

**Comment by EngrPips** [comment]
```js
function withdraw() public onlyOwner {
for (
uint256 funderIndex = 0;
funderIndex < s_funders.length;
funderIndex++
) {
address funder = s_funders[funderIndex];
s_addressToAmountFunded[funder] = 0;
}
s_funders = new address;
// Transfer vs call vs Send
// payable(msg.sender).transfer(address(this).balance);
(bool success, ) = i_owner.call{value: address(this).balance}("");
require(success);
}
```

**Reply by EngrPips** [reply, marked as answer]
Hey @0xhung295, isn't this line `s_funders = new address;` supposed to be the below?

```js
s_funders = new address[](0);
```

**Comment by KMean** [comment]
Oh so you're not on Html-fund-me-cu but on foundry-funde-me. As @EngrPips said the error is that you are trying to assign an address to an array. Solidity does not allow reinitialization of an array with a simple new address assignment. Instead, you should reset the array properly.

```solidity
function withdraw() public onlyOwner {
        for (uint256 funderIndex = 0; funderIndex < s_funders.length; funderIndex++) {
            address funder = s_funders[funderIndex];
            s_addressToAmountFunded[funder] = 0;
        }
        s_funders = new address[](0);
        // Transfer vs call vs Send
        // payable(msg.sender).transfer(address(this).balance);
        (bool success,) = i_owner.call{value: address(this).balance}("");
        require(success);
    }
```

If that works mark @EngrPips as answer.
